Lib/dis.py

```diff
@@ -66,6 +66,8 @@ def _try_compile(source, name):
     try:
         c = compile(source, name, 'eval')
     except SyntaxError:
+        c = None
+    if not c:
```

Comment:
Actually one more issue - it's not obvious to me that compile can never return None (even if it's true now, how do we know it will continue to be true?)

Maybe better to have a boolean variable, set it to ``False`` before the ``try:`` and to ``True`` in the ``except:``, and then check it in the if?

Reply:
Maybe I'm being silly, and it can't be None because it's supposed to be a code object or an ast. In that case maybe just change the check to "if c is not None", instead of relying on the bool() value of the result.

Reply:
For me it looks like call to `builtin_compile` always utilizes code object created by `_PyAssemble_MakeCodeObject` or AST returned by `PyAST_mod2obj` call, or sets exception and returns `NULL` if error is occurred, so its result can't be `None`.